src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_user_isOK() {
+        User expected = new User("BobGoo", "123456", 18);
+        registrationService.register(expected);
+        assertEquals(expected, storageDao.get(expected.getLogin()));
+    }
+
+    @Test
+    void register_userPasswordIsShort_isNotOK() {
+        User expected = new User("BobWithShortPassword", "12", 18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userLoginIsShort_isNotOK() {
+        User expected = new User("Bob", "1234567", 18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userUnderage_isNotOK() {
+        User expected = new User("BobChild", "1234567", 15);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
```

Comment:
You should test all possible edge cases. For instance, you should test an age of 17 (which is not OK), and an age of 18 (which is OK). This will ensure that your program behaves correctly at the boundaries of your conditions.